vassal-app/src/main/java/VASSAL/build/module/Map.java

```diff
@@ -911,6 +911,13 @@ public Point snapTo(Point p) {
     snap.translate(-r.x, -r.y);
     snap = b.snapTo(snap);
     snap.translate(r.x, r.y);
+
+    //CC bugfix13409
+    // If we snapped to a point outside the board b, call sanpTo again with the board we landed into
+    final Board bSnappedTo = findBoard(snap);
+    if (!b.equals(bSnappedTo)) {
+      snap = this.snapTo(snap);
```

Comment:
This should be snap  = bSnappedTo.snapTo(snap), not snap = this.snapTo(snap). We already know which board we need to do the re-snap, no need to recurse.